### PR TITLE
Style album results

### PR DESCRIPTION
**Pull request by mDevCodes**

### CODE REVIEW

File: app/utils/discogs.schemas.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import z from "zod";
 
 const AlbumSchema =  z.object({
+  cover_image:z.string(),
+  title: z.string().optional(),
   year: z.string().optional(),
-  id: z.number().optional()
+  id: z.number().optional(),
```
Comment (m1yon) [inline app/utils/discogs.schemas.ts]:
```suggestion
  id: z.number()
```

this should actually always be defined since it's a unique identifier for the record. If they aren't always defined from the discogs API, I would generate one

File: app/utils/discogs.schemas.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import z from "zod";
 
 const AlbumSchema =  z.object({
+  cover_image:z.string(),
```
Comment (m1yon) [inline app/utils/discogs.schemas.ts]: properties on objects should almost always be camelCase

```suggestion
  coverImage: z.string(),
```

File: app/utils/discogs.schemas.ts [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
+  title: z.string().optional(),
```
Comment (m1yon) [inline app/utils/discogs.schemas.ts]: title should always be defined yeah?

```suggestion
  title: z.string(),
```

File: app/utils/discogs.schemas.ts [hunk 4]
```diff
@@ -1,8 +1,10 @@
...
   year: z.string().optional(),
```
Comment (m1yon) [inline app/utils/discogs.schemas.ts:7]: same with year?

```suggestion
  year: z.string(),
```

File: next.config.js [hunk 5]
```diff
@@ -1,4 +1,19 @@
+const { hostname } = require("os");
```
Comment (m1yon) [inline next.config.js]:
```suggestion
```

don't need this, I'm guessing it was an accidental auto-import

File: app/utils/discogs.schemas.ts (hunk 2 above)
Comment (m1yon) [reply app/utils/discogs.schemas.ts]: if discogs isn't returning it to you this way, I'd adjust it in a .map (you can massage the data to match how **you** want it to loo)

File: components/Result.tsx [hunk 6]
```diff
@@ -1,9 +1,19 @@
 import { Album } from "@/app/utils/discogs.schemas";
+import Image from "next/image";
 
 export default function Result({ album }: { album: Album }) {
   return (
-    <>
-      <h3>Year: {album.year}</h3>
-    </>
+    <div className="flex mt-16">
```
Comment (m1yon) [inline components/Result.tsx]: you can definitely go a little harder on the styles here if you want. Since the PR is for styling the album results, I'd go ahead and get these styles pretty close to how you want them to look in the end (not including the buttons of course).

**Review by m1yon — CHANGES_REQUESTED** [review]

File: app/utils/discogs.schemas.ts (hunk 1 above)
Comment (mDevCodes) [reply app/utils/discogs.schemas.ts]: Tested and there don't seem to be cases where there is no Id

File: app/utils/discogs.schemas.ts (hunk 3 above)
Comment (mDevCodes) [reply app/utils/discogs.schemas.ts]: Tested and there don't seem to be cases where there is no title

File: next.config.js (hunk 5 above)
Comment (mDevCodes) [reply next.config.js]: Removed, not sure where this came from

File: app/utils/discogs.schemas.ts (hunk 2 above)
Comment (mDevCodes) [reply app/utils/discogs.schemas.ts]: Modified the key but getting a type error for album in my map method in discogs. Tried to assign type AlbumSchema to album but cover_image doesn't exist on it now so that throws an error. This is also an example of a time where I don't understand why I have to explicitly define the type of album where as in other cases I don't.

File: app/utils/discogs.schemas.ts (hunk 4 above)
Comment (mDevCodes) [reply app/utils/discogs.schemas.ts:7]: year is not defined for every object. Some objects are missing the year key value pair. I left this as optional and inserted a ternary operator in Result to handle the undefined case.

File: components/Result.tsx (hunk 6 above)
Comment (mDevCodes) [reply components/Result.tsx]: I have updated some styles and will continue to do so throughout this PR.

File: app/test.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { expect, test } from "vitest";
```
Comment (m1yon) [inline app/test.ts]: this should be in a file named `splitTitle.test.ts` and be in the same directory as `splitTitle.ts`

File: app/test.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+import splitTitle from "./utils/splitTitle";
+
+test("splits title into artist and album", () => {
```
Comment (m1yon) [inline app/test.ts]: nicee, you can add other edge cases as additional tests if you run across them

File: app/utils/discogs.ts [hunk 9]
```diff
@@ -1,26 +1,29 @@
...
+
     // GET call to Discogs API
     const result = await fetch(url);
     const data = await result.json();
-  
-    // schema check using Zod
-    SearchResponseSchema.parse(data);
+
+      // schema check using Zod
+      SearchResponseSchema.parse(data);
```
Comment (m1yon) [inline app/utils/discogs.ts:24]: the return value from this is what you want to return since Zod won't include fields that you don't specify in your schema

File: app/utils/splitTitle.ts [hunk 10]
```diff
@@ -0,0 +1,6 @@
+export default function splitTitle(title: string) {
+  const titleSplit = title.split("-");
+  const artist = titleSplit[0].trimEnd();
```
Comment (m1yon) [inline app/utils/splitTitle.ts:3]: nicee, good way of doing it

File: app/utils/splitTitle.ts [hunk 11]
```diff
@@ -0,0 +1,6 @@
...
+  const album = titleSplit[1].trimStart();
+  return [artist, album];
```
Comment (m1yon) [inline app/utils/splitTitle.ts]: I'd return this as an object, that way when you call this you know what the return values actually are.

I'd only return a tuple if it made sense for the destructed variable names to be different (like `useState` for example since the destructured name we want is dependent on the context)

```suggestion
  return { artist, album };
```

File: components/Result.tsx [hunk 12]
```diff
@@ -1,9 +1,33 @@
 import { Album } from "@/app/utils/discogs.schemas";
+import Image from "next/image";
+import Icon from "./Icon";
 
 export default function Result({ album }: { album: Album }) {
   return (
-    <>
-      <h3>Year: {album.year}</h3>
-    </>
+    album.title && (
```
Comment (m1yon) [inline components/Result.tsx]: I'd use `album.title ? <div>...</div> : null` here

[here's a quick article on why](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx)

File: components/Result.tsx [hunk 13]
```diff
@@ -1,9 +1,33 @@
...
+            className="rounded-sm"
+            src={album.cover_image}
+            alt={`${album.title} cover image`}
+            width={200}
+            height={200}
+          />
+        )}
+        <div className="ml-14">
+          <h3 className="text-lg font-bold">Title: {album.title}</h3>
+          <h3>Year: {album.year ? album.year : "-"}</h3>
```
Comment (m1yon) [inline components/Result.tsx]: this isn't really a heading

```suggestion
          <p>Year: {album.year ? album.year : "-"}</p>
```

Comment (m1yon) [inline components/Result.tsx]: also probably don't need to include a year label since it's implied, your choice though

```suggestion
          <h3>{album.year ? album.year : "-"}</h3>
```

**Review by m1yon — CHANGES_REQUESTED** [review]

File: app/test.ts (hunk 7 above)
Comment (mDevCodes) [reply app/test.ts]: That's hilarious. I initially named this file with that convention but figured it was best to keep it general and include all of my tests in one file. I just re-ran my test file with the "test.js" name and it failed. Changed it to this convention and it worked again. Why is that?

File: components/Result.tsx (hunk 12 above)
Comment (mDevCodes) [reply components/Result.tsx]: Makes sense, great resource.

File: components/Result.tsx (hunk 13 above)
Comment (mDevCodes) [reply components/Result.tsx]: Agreed, without the title is a lot cleaner

**Review by m1yon — APPROVED** [review]
LGTM
